File: terraform/lambdas/shared/response.py
import json
from typing import Any
from decimal import Decimal

class DecimalEncoder(json.JSONEncoder):
    """Custom JSON encoder that converts Decimal to float"""
    def default(self, obj):
        if isinstance(obj, Decimal):
            return float(obj)
        return super(DecimalEncoder, self).default(obj)

def error_response(status_code: int, code: str, message: str, request_id: str) -> dict:
    return {
        'statusCode': status_code,
        'headers': {'Content-Type': 'application/json'},
        'body': json.dumps({
            'error': {
                'code': code,
                'message': message,
                'requestId': request_id
            }
        })
    }

def success_response(status_code: int, data: Any) -> dict:
    return {
        'statusCode': status_code,
        'headers': {'Content-Type': 'application/json'},
        'body': json.dumps(data, cls=DecimalEncoder)
    }

File: terraform/lambdas/shared/response.ts
import type { APIGatewayProxyResultV2 } from "aws-lambda";
import type { ErrorResponse } from "./types.js";

/**
 * Create a standardized error response
 */
export function errorResponse(
  statusCode: number,
  code: string,
  message: string,
  requestId: string
): APIGatewayProxyResultV2 {
  const response: ErrorResponse = {
    error: {
      code,
      message,
      requestId,
    },
  };
  return {
    statusCode,
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(response),
  };
}

/**
 * Create a standardized success response
 */
export function successResponse(statusCode: number, data: any): APIGatewayProxyResultV2 {
  return {
    statusCode,
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(data),
  };
}



